libcontainerd/client_linux.go

```diff
@@ -98,13 +98,9 @@ func (clnt *client) AddProcess(ctx context.Context, containerID, processFriendly
 
 	container.processes[processFriendlyName] = p
 
-	clnt.unlock(containerID)
-
-	if err := clnt.backend.AttachStreams(processFriendlyName, *iopipe); err != nil {
-		clnt.lock(containerID)
+	if err := attachStdio(*iopipe); err != nil {
```

Comment:
Could change that. It's so because it is easier to return nil instead empty struct in `openFifos` and I didn't want to pass pointers through pkg boundary when the other side is not meant to change the struct.